src/connection/socket.ts

```diff
@@ -94,11 +94,21 @@ export class RethinkDBSocket extends EventEmitter {
     const { tls = false, ...options } = this.connectionOptions;
     try {
       const socket = await new Promise<Socket>((resolve, reject) => {
-        const s = tls
-          ? tlsConnect(options)
-          : netConnect(options as TcpNetConnectOpts);
-        s.once('connect', () => resolve(s)).once('error', reject);
+        if (tls) {
+          let socket = tlsConnect(options);
+          socket.once('secureConnect', () => {
+            if (socket.authorized) {
```

Comment:
It is better to throw an error is `authorized` field is not true for any reasons

Reply:
Done. The error handler should probably always be called in case of an authorization error.

Reply:
I've also added a `socket.destroy` call to the error path, although the socket seems to close on it's own for some reason? More importantly, I've added a check for `rejectUnauthorized`, since it has no bearing on the `authorized` property and just prevents an `error` event from being emited.